Objects/unionobject.c

```diff
@@ -490,10 +490,10 @@ _Py_Union(PyObject *args)
     }
 
     result->args = dedup_and_flatten_args(args);
+    _PyObject_GC_TRACK(result);
     if (result->args == NULL) {
-        PyObject_GC_Del(result);
```

Comment:
@Fidget-Spinner For next occasions, the problem with this is that `PyObject_GC_Del` cannot be called like that because it overrides a bunch of cleanups like the call to `_Py_ForgetReference` and other possible handling.

Reply:
Sorry bout that. I'll keep that in mind in the future.

BTW, does this mean this should be changed as well? https://github.com/python/cpython/blob/main/Objects/genericaliasobject.c#L654
I'm confused as to why it doesn't break the buildbots, or maybe because none of our tests make the object setup fail so ``PyObject_GC_Del`` is never called?

Reply:
Yes, that has exactly the same problem. I assume is not being reaches by the tests and that's why we are not seeing corruption.